migrations/007-source-projects/migrate.py

```
@@ -0,0 +1,981 @@
+                        self.municipality_lookup[key_ne_norm] = loc
+
+        self.context.log(f"Built location lookups: {len(self.location_lookup)} entries")
+
+    async def _build_organization_cache(self) -> None:
+        """Build cache of existing organizations."""
+        orgs = await self.context.db.list_entities(
+            entity_type="organization", limit=10_000
+        )
+        for org in orgs:
+            for name in org.names:
+                if name.en and name.en.full:
+                    key = name.en.full.strip().lower()
+                    self.organization_cache[key] = org.id
+                if name.ne and name.ne.full:
+                    key = name.ne.full.strip().lower()
+                    self.organization_cache[key] = org.id
+
+        self.context.log(
+            f"Built organization cache: {len(self.organization_cache)} entries"
+        )
+
+    async def _migrate_projects(self) -> None:
+        """Import projects from the pre-transformed JSONL file."""
+        # Load projects from JSONL file
+        projects = []
+        source_file = self.context.migration_dir / "source" / "dfmis_projects.jsonl"
+        try:
+            with open(source_file, "r", encoding="utf-8") as f:
+                for line in f:
+                    line = line.strip()
+                    if line:
+                        projects.append(json.loads(line))
+            self.context.log(
+                f"Loaded {len(projects)} projects from source/dfmis_projects.jsonl"
+            )
+        except FileNotFoundError:
+            self.context.log("ERROR: source/dfmis_projects.jsonl not found.")
+            self.context.log(
+                "Please run the scraper first: cd migrations/007-source-projects/mof_dfmis && python scrape_mof_dfmis.py"
+            )
+            raise
+
+        if not projects:
+            self.context.log("WARNING: No projects found in source data.")
+            return
```

return or raise?

---

done - now it raises a ValueError instead of silently returning.